events/events.go

```diff
@@ -50,24 +53,51 @@ func ListenForEvents(ctx context.Context, config Config) {
 	}
 }
 
+// listenToPostgresNotify listens to postgres notifications.
+// It will retry on failure for dbReconnectMaxAttempt times.
 func listenToPostgresNotify(ctx context.Context, pgNotify chan bool) {
-	conn, err := db.Pool.Acquire(ctx)
-	if err != nil {
-		logger.Errorf("error creating database pool: %v", err)
-	}
+	var reconnectAttempts int
 
-	_, err = conn.Exec(ctx, "LISTEN event_queue_updates")
-	if err != nil {
-		logger.Errorf("error listening to database notify: %v", err)
+	var listenToPostgresNotify = func(ctx context.Context, pgNotify chan bool) error {
+		conn, err := db.Pool.Acquire(ctx)
+		if err != nil {
+			return fmt.Errorf("error acquiring database connection: %v", err)
+		}
+		defer conn.Release()
+
+		reconnectAttempts = 0
+		logger.Infof("acquired database connection")
+
+		_, err = conn.Exec(ctx, "LISTEN event_queue_updates")
+		if err != nil {
+			return fmt.Errorf("error listening to database notifications: %v", err)
+		}
+		logger.Infof("listening to database notifications")
+
+		for {
+			notification, err := conn.Conn().WaitForNotification(ctx)
+			if err != nil {
+				return fmt.Errorf("error listening to database notifications: %v", err)
+			}
+
+			logger.Debugf("Received database notification: %+v", notification)
+			pgNotify <- true
+		}
 	}
 
+	// retry on failure.
 	for {
-		_, err = conn.Conn().WaitForNotification(ctx)
-		if err != nil {
-			logger.Errorf("error waiting for database notifications: %v", err)
+		if err := listenToPostgresNotify(ctx, pgNotify); err != nil {
+			reconnectAttempts++
+			logger.Errorf("error listening to database notifications: %v", err)
+		}
+
+		if reconnectAttempts > dbReconnectMaxAttempt {
+			logger.Fatalf("failed to reconnect to database after %d attempts", dbReconnectMaxAttempt)
```

Comment:
Fatalf will stop the process which we never want, 

I think use something https://github.com/sethvargo/go-retry, with an ExponentialBack of 1s with a MaxDuration of  5 minutes. 

We would then only log an error message every 5 minutes.